File: 1213-intersection-of-three-sorted-arrays/1213-intersection-of-three-sorted-arrays.py
class Solution:
    def arraysIntersection(self, arr1: List[int], arr2: List[int], arr3: List[int]) -> List[int]:
        s1 = set(arr1)
        s2 = set(arr2)
        s3 = set(arr3)
        return sorted(list(s1 & s2 & s3))

File: 1213-intersection-of-three-sorted-arrays/1213-intersection-of-three-sorted-arrays.cpp
class Solution {
public:
    vector<int> arraysIntersection(vector<int>& arr1, vector<int>& arr2, vector<int>& arr3) {
        map<int, int> m;
        for(int i: arr1) m[i]++;
        for(int i: arr2) m[i]++;
        for(int i: arr3) m[i]++;
        
        vector<int> ans;
        for (auto& p: m)
            if (p.second == 3)
                ans.push_back(p.first);
        sort(ans.begin(), ans.end());
        return ans;
    }
};